### ISSUE TITLE
esp-idf V3.0 how to config the component.mk file ?

### ISSUE THREAD

**Issue by 1XiaSheng** [espressif/ESP8266_RTOS_SDK#371]
![tim 20181105155331](https://user-images.githubusercontent.com/37097029/47984913-076cd000-e113-11e8-92fd-42f23a89bf52.png)
![tim 20181105155338](https://user-images.githubusercontent.com/37097029/47984919-0b98ed80-e113-11e8-8656-478f52e139b7.png)

**Comment by 1XiaSheng** [comment]
Great God, please see the above figure, what configuration is required for each component.mk file in the directory structure?

The correct guidelines are not available at https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/build-system.html.

大神,请看上面的图,目录结构中的各个component.mk文件需要填入什么配置?

在https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/build-system.html 中无法得到正确指引.

**Comment by 27-301** [comment]
for folder structure, you can refer to https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/build-system.html#example-project, you can rename app to components, for each component.mk in usr/hdw/cfg, you can refer to SDK/components/jsmn/component.mk.

**Comment by 1XiaSheng** [comment]
I hope to explain the specific example in the above picture, otherwise I still don't know how to add it.thanks
我希望是以上图中的具体的例子来说明,否则还是不知道怎么添加.谢谢

**Comment by 27-301** [comment]
you can upload your example here. But hope you can read the document and refer to components in SDK as example.

**Comment by donghengqaz** [comment]
在你的 Makfile 里面添加:
```
EXTRA_COMPONENT_DIRS := $(shell pwd)/app
```
去掉 app 目录里面的 component.mk

**Comment by 1XiaSheng** [comment]
[图片]

**Comment by 1XiaSheng** [comment]
![tim 20181107181734](https://user-images.githubusercontent.com/37097029/48125455-afbc9900-e2b9-11e8-8e23-3a8eba74f6c8.png)
![tim 20181107181853](https://user-images.githubusercontent.com/37097029/48125463-b3502000-e2b9-11e8-92f8-e2c8a4b99dd4.png)

按这个结构配置成功.

**Comment by jhinklejr** [comment]
1XiaSheng

Unless you are doing something advanced for the files specific, leave them empty.

An example of when or how to use them -- I was having a fault, and wanted to use a mix c and assembler listing to match the exception address to an exact line of code.

I added a single line, to that specific make file, which added extra CFLAG settings.  I removed it when I was done.

I've been doing a lot of advanced work and have yet, other than the example above, to make any modifications to the component make file.

They are there, in the default make process, to use if necessary.

Hope that helps
